src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -305,8 +295,13 @@ public class XmlConvert
         }
 
         private const int EncodedCharLength = 7; // ("_xFFFF_".Length);
-        private static volatile Regex? s_encodeCharPattern;
-        private static volatile Regex? s_decodeCharPattern;
+
+        [RegexGenerator("_[Xx][0-9a-fA-F]{4}(?:_|[0-9a-fA-F]{4}_)")]
```

Comment:
The change does look fine for this one as Dan said, but if you changed the below to factor out the 4 repeater because it was more efficient, why did you go the other way around with this one? If the previous pattern was more efficient, shouldn't it have been left as it was before?

Reply:
I don't understand.  Both this DecodeCharRegex and the EncodeCharRegex were changed from:
```
([0-9a-fA-F]{4}|[0-9a-fA-F]{8})_
```
to
```
[0-9a-fA-F]{4}(?:_|[0-9a-fA-F]{4}_)
```
Can you elaborate?

Reply:
Nvm I was looking at it backwards for some reason I was seeing that in the DecodeCharRegex you were going from a 4 factored out repeater to a 8 repeater but I think it was just my brain playing tricks on me. I think the change here makes sense for both.